src/login.tsx

```diff
@@ -0,0 +1,231 @@
+    const logo = new LogInIcon(router);
+    shell.add(logo, 'top', { rank: 10000 });
+  }
+};
+
+export default userIcon;
+
+class LogInIcon extends ReactWidget {
+  constructor(router: IRouter) {
+    super();
+    this.id = 'jp-MainLogo';
+    this.title.label = 'LogIn';
+    this.title.caption = 'LogIn with GitHub';
+
+    this._router = router;
+    this._profile = {
+      login: "Anonymous"
+    };
+  }
+
+  protected onAfterAttach(msg: Message): void {
+    super.onAfterAttach(msg);
+    window.addEventListener('click', this._onClickOutSide);
+
+    requestAPI<any>('user').then(data => {
+      console.debug("Data: ", data);
+      if (data.anonymous) {
+        InputDialog.getText({
+          title: "User",
+          label: "Who are you?",
+          okLabel: "Save",
+          text: getAnonymousUserName()
+        }).then( value => {
+          console.debug("Value:", value);
+          const username = value.value.split(' ');
+          let name = username[0].substring(0, 1).toLocaleUpperCase();
+          if (username.length > 1) {
+            name += username[1].substring(0, 1).toLocaleUpperCase();
+          }
+
+          // send name to the server
+          const init = {
+            method: 'POST',
+            body: '{"name": "' + value.value + '"}'
+          };
+          requestAPI<any>('anonymous', init);
+
+          this._profile = {
+            login: value.value,
+            avatar: name,
+            color: getRandomColor()
+          };
+          this.update();
+        });
+      } else {
+        this._profile = data;
+        this.update();
+      }
+    });
+  }
+
+  private _onClickOutSide = (e: MouseEvent): void => {
+    if (!this.node.contains(e.target as Node) && this._isActive) {
+      this._isActive = false;
+      this.update();
+    }
+  };
+
+  private _onClick = (): void => {
+    this._isActive = !this._isActive;
+    this.update();
+  };
+
+  private _logIn = () => {
+    this._router.navigate('/login', { hard: true });
+  };
+
+  private _logOut = () => {
+    this._router.navigate('/logout', { hard: true });
+  };
+
+  render(): React.ReactElement {
+    console.debug(this._profile);
+    const getAvatar = () => {
+      return (
+        <div className="login-container">
+          <div
+            onClick={this._onClick}
+            className="login-icon"
+            style={{backgroundColor: this._profile.color}}
+          >
+            <span>{this._profile.avatar}</span>
+          </div>
+          <div
+            className={`login-menu ${this._isActive ? 'active' : 'inactive'}`}
+          >
+            <ul>
+              <li key="Anonymous">
+                <div><span>Logged in as {this._profile.login}</span></div>
+              </li>
+              <hr />
+              <li key="login" className="login-menu-clickable">
+                <div onClick={() => this._logIn()}>
+                  <span>Log in</span>
+                </div>
+              </li>
+            </ul>
```

Comment:
Small suggestion here. I think it will be good to use a Lumino menu for this to have a better UI integration.

> This means creating commands for those actions.